API/controllers/entityController.go

```diff
@@ -506,7 +508,130 @@ func HandleGenericObjects(w http.ResponseWriter, r *http.Request) {
+//     If none provided, all namespaces are used by default.'
+//   - name: fieldOnly
+//     in: query
+//     description: 'specify which object field to show in response.
+//     Multiple fieldOnly can be added. An invalid field is simply ignored.'
+//   - name: startDate
+//     in: query
+//     description: 'filter objects by lastUpdated >= startDate.
+//     Format: yyyy-mm-dd'
+//   - name: endDate
+//     in: query
+//     description: 'filter objects by lastUpdated <= endDate.
+//     Format: yyyy-mm-dd'
+//   - name: attributes
+//     in: query
+//     description: 'Any other object attributes can be queried.
+//     Replace attributes here by the name of the attribute followed by its value.'
+//     required: false
+//     type: string
+//     default: domain=DemoDomain
+//     example: vendor=ibm ; name=siteA ; orientation=front
+//   - name: filter
+//     in: body
+//     description: A JSON containing a mongoDB query to select and filter the desired objects.
+//     Operators can be `$not`, `$lt`, `$lte`, `$gt`, `$gte`, `$and` and `$or`.
+//     For equality, the syntax is: `[field]: value`.
+//     Objects can be filtered by any of their properties and attributes.'
+//     required: true
+//     type: JSON
+//     example: {"$and": [{"domain": "DemoDomain"}, {"attributes.height": {"$lt": "3"}}]}
+// responses:
+//		'200':
+//		    description: 'Found. A response body will be returned with
+//	        a meaningful message.'
+//     '400':
+//         description: 'Bad request. Request has wrong format.'
+//		'500':
+//		    description: Internal Error. A system error stopped the request.
 
+func HandleComplexFilters(w http.ResponseWriter, r *http.Request) {
+	fmt.Println("******************************************************")
+	fmt.Println("FUNCTION CALL: 	 HandleComplexFilters ")
+	fmt.Println("******************************************************")
+	DispRequestMetaData(r)
+	var complexFilters map[string]interface{}
+	matchingObjects := []map[string]interface{}{}
+
+	// Get user roles for permissions
+	user := getUserFromToken(w, r)
+	if user == nil {
+		return
+	}
+
+	err := json.NewDecoder(r.Body).Decode(&complexFilters)
+	if err != nil {
+		w.WriteHeader(http.StatusBadRequest)
+		u.Respond(w, u.Message("Error while decoding request body"))
+		u.ErrLog("Error while decoding request body", "HANDLE COMPLEX FILTERS", "", r)
+		return
+	}
+	utils.ApplyWildcardsOnComplexFilter(complexFilters)
+
+	// Get objects
+	filters := getFiltersFromQueryParams(r)
+	req := u.FilteredReqFromQueryParams(r.URL)
+	entities := u.GetEntitiesByNamespace(filters.Namespace, filters.Id)
+
+	for _, entStr := range entities {
+		// Get objects
+		entData, err := models.GetManyObjectsComplex(entStr, req, filters, complexFilters, user.Roles)
+		if err != nil {
+			u.ErrLog("Error while looking for objects at "+entStr, "HandleComplexFilters", err.Message, r)
+			u.RespondWithError(w, err)
+			return
+		}
+
+		// Save entity to help delete and respond
```

Comment:
Is this copied code useful here?

Reply:
It is now that the `DELETE` method has been added to the `HandleComplexFilters` endpoint.